cmd/snap/cmd_snapshot.go

```diff
@@ -0,0 +1,303 @@
+	if len(sg.Snapshots) == 0 {
+		return fmt.Sprintf("%d\t-\t-\t-", sg.ID)
+	}
+	mint := sg.MinTime()
+	snapNames := make([]string, 0, len(sg.Snapshots))
+	var sz int64
+	for _, sh := range sg.Snapshots {
+		snapNames = append(snapNames, sh.Snap)
+		sz += sh.Size
+	}
+	snaps := strings.Join(snapNames, ", ")
+	if !x.Wide {
+		width, _ := termSize()
+		// size (5) + gutters (3+2+2; why is the first gutter 3?)
+		width -= 12 + extraWidth
+		if len(snaps) > width {
+			snaps = snaps[:width-1] + "…"
+		}
+	}
+	return fmt.Sprintf("%d\t%s\t%s\t%s", sg.ID, x.fmtTime(mint), snaps, fmtSize(sz))
+}
+
+func (x *savedCmd) Execute([]string) error {
+	list, err := Client().Snapshots(uint64(x.ID), installedSnapNames(x.Positional.Snaps))
+	if err != nil {
+		return err
+	}
+	if len(list) == 0 {
+		fmt.Fprintln(Stdout, "No snapshots found.")
+		return nil
+	}
+	w := tabWriter()
+	defer w.Flush()
+
+	fmt.Fprintln(w, "ID\tDate\tSnaps\tSize")
+	// the list is ordered by id
+	minTimeWidth := len(x.fmtTime(list[0].MinTime()))
+	maxIDwidth := len(strconv.FormatUint(list[len(list)-1].ID, 10))
+	extraWidth := maxIDwidth + minTimeWidth
+
+	for _, sg := range list {
+		fmt.Fprintln(w, x.tabline(&sg, extraWidth))
+	}
+
+	return nil
+}
+
+type saveCmd struct {
+	waitMixin
+	timeOpts
+	Homes      []string `long:"homes"`
+	Positional struct {
+		Snaps []installedSnapName `positional-arg-name:"<snap>"`
+	} `positional-args:"yes"`
+}
+
+var shortSaveHelp = i18n.G("Save a snapshot of the current data")
+var longSaveHelp = i18n.G(`
+The save command creates a snapshot of the current data for the given snaps.
+`)
+
+func (x *saveCmd) Execute([]string) error {
+	cli := Client()
+	changeID, err := cli.SnapshotMany(installedSnapNames(x.Positional.Snaps), x.Homes)
+	if err != nil {
+		return err
+	}
+	chg, err := x.wait(cli, changeID)
+	if err == noWait {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	var shID snapshotID
+	chg.Get("snapshot-id", &shID)
+	y := &savedCmd{
+		timeOpts: x.timeOpts,
+		ID:       shID,
+	}
+	y.Positional.Snaps = x.Positional.Snaps
+	return y.Execute(nil)
+}
+
+type loseCmd struct {
+	waitMixin
+	Positional struct {
+		ID    snapshotID          `positional-arg-name:"<id>"`
+		Snaps []installedSnapName `positional-arg-name:"<snap>"`
+	} `positional-args:"yes" required:"yes"`
+}
+
+var shortLoseHelp = i18n.G("Delete a snapshot")
+var longLoseHelp = i18n.G(`
+The lose command deletes a snapshot.
+`)
+
+func (x *loseCmd) Execute([]string) error {
+	cli := Client()
+	snaps := installedSnapNames(x.Positional.Snaps)
+	changeID, err := cli.LoseSnapshot(uint64(x.Positional.ID), snaps)
+	if err != nil {
+		return err
+	}
+	_, err = x.wait(cli, changeID)
+	if err == noWait {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	if len(snaps) > 0 {
+		// TRANSLATORS: the %s is a comma-separated list of quoted snap names
+		fmt.Fprintf(Stdout, i18n.G("Snapshot #%d of snaps %s is lost to the ages.\n"), x.Positional.ID, strutil.Quoted(snaps))
```

Comment:
These messages feel a bit too enigmatic. It's not entirely clear if it worked or not. :)

I suggest something terse like "Snapshot %d discarded." or similar.

Reply:
I forgot to change this one (and the `*fine*` one about check) before committing... was having fun playtesting :-D

Reply:
I think what we usually try to do is, when the user says `snap verb noun`, we respond with `Noun verbed.` on success:
```console
$ snap remove eeevil 
eeevil removed
```

in keeping with this, which I think is a good practice, and assuming we go with `forget` for this verb, we'd have
```console
$ snap forget 42
Snapshot 42 forgotten.
```
If you think the response should be `Snapshot 42 discarded`, this means you're really thinking that the verb should be `discard`...